services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentService.kt

```diff
@@ -58,14 +61,12 @@ class ApplicationAttachmentService(
         attachmentType: ApplicationAttachmentType,
         attachment: MultipartFile
     ): ApplicationAttachmentMetadata {
-        val application = findApplication(applicationId)
-
-        if (!isPending(application)) {
-            logger.warn { "Application is processing, cannot add attachment." }
-            throw ApplicationAlreadyProcessingException(application.id, application.alluid)
-        }
-
-        validateAttachment(attachment)
+        val application =
+            findApplication(applicationId).also { application ->
+                checkCondition { pending(application) }
+                checkCondition { roomForAttachment(applicationId) }
+                checkCondition { validFile(attachment) }
+            }
```

Comment:
I like the idea of collecting the different condition checks into a single block with simple method calls for the checks. Though I think `checkCondition` is superfluous since it does nothing. Also, the methods should be named so it's clear what they're used for.

```suggestion
            findApplication(applicationId).also { application ->
                ensurePending(application)
                ensureRoomForAttachment(applicationId)
                validateFile(attachment)
            }
```

Names like `pending()` are vague in what they're used for. Does it return a pending version of the application? Does it return a boolean on whether the application is considered pending? Or does it do something else, like here?

Reply:
I originally decided to keep it to improve readability of this code block. But as it is somewhat confusing and not bringing any added value, I'll remove it.

Reply:
checkCondition is now removed.